### DISCUSSION TITLE
Get computed values from Chrome browser

### DISCUSSION THREAD

**Discussion by Karnikel** [cypress-io/cypress#9529] [Questions and Help]
If I observe a DOM Element manually in Chromium browser, there is a possibility to look at computed values. 
Is it also possible to get them via Cypress to use it for assertions or as values for further functionality?

In my case, I need the hight and width of a DOM Element to calculate touch gesture coordinates.
Hint: 
To observe computed values: Open DevTools (F12) -> Select Tab "Elements" -> Select Tab "Computed" on the other view

**Comment by portyu9** [comment]
Hi @Karnikel 

Yes. Cypress can access the same computed DOM values that Chrome DevTools shows.

For rendered width/height and coordinates, use `getBoundingClientRect()`:

```js
cy.get('.my-element').then(($el) => {
  const rect = $el[0].getBoundingClientRect()

  expect(rect.width).to.be.greaterThan(0)
  expect(rect.height).to.be.greaterThan(0)

  const centerX = rect.left + rect.width / 2
  const centerY = rect.top + rect.height / 2
})
```

If you specifically need a computed CSS property:

```js
cy.get('.my-element').then(($el) => {
  const styles = getComputedStyle($el[0])

  expect(styles.width).to.equal('200px')
  expect(styles.height).to.equal('100px')
})
```

For calculating touch-gesture coordinates, prefer `getBoundingClientRect()` because it returns the element's actual rendered position and dimensions.

**Reply by portyu9** [reply]
@Karnikel Please mark this answer as accepted if it helped, thank you.
